Mathlib/AlgebraicGeometry/EllipticCurve/Affine/Point.lean

```diff
@@ -487,6 +487,43 @@ lemma zero_def : 0 = (.zero : W'.Point) :=
 lemma some_ne_zero {x y : R} (h : W'.Nonsingular x y) : Point.some h ≠ 0 := by
   rintro (_ | _)
 
+/-- The `X` coordinate of a given point. For the point at infinity, this returns `0`
+(junk value). -/
+def x : W'.Point → R
+  | 0 => 0
+  | @some _ _ _ x _ _ => x
+
+/-- The `Y` coordinate of a given point. For the point at infinity, this returns `0`
+(junk value). -/
+def y : W'.Point → R
+  | 0 => 0
+  | @some _ _ _ _ y _ => y
+
+/-- Custom recursor for `[NeZero P]`. -/
+@[elab_as_elim]
+def neZeroRec {C : W'.Point → Sort*} (h₁ : {x y : R} → (h : W'.Nonsingular x y) → C (some h)) :
+    (P : W'.Point) → [NeZero P] → C P
+  | 0, h₀ => (h₀.ne _ rfl).elim
+  | some h, _ => h₁ h
+
+lemma equation_x_y (P : W'.Point) [NeZero P] : W'.Equation P.x P.y :=
+  P.neZeroRec fun h ↦ h.1
+
+lemma equation_x_y' (P : W'.Point) [NeZero P] : P.y ^ 2 + W'.a₁ * P.x * P.y + W'.a₃ * P.y
+    = P.x ^ 3 + W'.a₂ * P.x ^ 2 + W'.a₄ * P.x + W'.a₆ :=
```

Comment:
IIRC, the `=` is supposed to stay on the previous line.
(I'd also prefer the statement to start in a new line when it spans more than one line, but this is a matter of taste.)